### ISSUE TITLE
feat: cross-subject generalization toolkit (IAF, alignment, SSL, augmentation)

### ISSUE THREAD

**Issue by rajnavakoti** [neural-state-lab/neuro-state-monitor#35]
## Description
Implement all cross-subject generalization techniques as pluggable framework components. Users enable what they need via config. The cross-subject problem is the #1 limitation in EEG ML — our framework should offer every established solution.

## Current State
- ✅ Subject-based splitting (honest measurement)
- ✅ Normalized band power (removes amplitude differences)
- ✅ Specparam (removes 1/f slope differences)
- ❌ Individual Alpha Frequency adjustment
- ❌ Euclidean Alignment
- ❌ Self-supervised pre-training
- ❌ Data augmentation for EEG
- ❌ Domain adaptation

### 1. Individual Alpha Frequency (IAF) Adjustment
**Difficulty:** Easy | **Impact:** Medium
```python
# Instead of fixed alpha = 8-13 Hz for everyone:
iaf = compute_individual_alpha_frequency(epochs)  # e.g., 9.5 Hz
alpha_band = (iaf - 2, iaf + 2)  # personalized: 7.5-11.5 Hz
```
- [ ] `shared/features/iaf.py` — detect IAF per subject from resting EEG or peak detection
- [ ] Update SpectralConfig to accept `use_iaf=True` for personalized bands
- [ ] Falls back to fixed bands if IAF detection fails

### 2. Euclidean Alignment
**Difficulty:** Medium | **Impact:** Medium-High
```python
# Align covariance matrices across subjects before feature extraction
aligned_epochs = euclidean_align(epochs, subjects)
```
- [ ] `shared/preprocessing/alignment.py` — Euclidean Alignment implementation
- [ ] Option to apply in preprocessing pipeline via config: `use_alignment=True`
- [ ] Compatible with MOABB's built-in implementation

### 3. Self-Supervised Pre-Training (selfEEG)
**Difficulty:** Hard | **Impact:** High (biggest accuracy boost)
```python
# Pre-train on large unlabeled EEG, fine-tune on small labeled
pretrained = ssl_pretrain(unlabeled_epochs, method="simclr")
model = fine_tune(pretrained, labeled_epochs, labels)
```
- [ ] `shared/models/ssl_pretrain.py` — wrapper around selfEEG (SimCLR/BYOL)
- [ ] Script to pre-train on combined UCI + PEERS + OpenNeuro data
- [ ] Pre-trained weights saveable/loadable for reuse
- [ ] Integration with training scripts as optional backbone

### 4. EEG Data Augmentation (TorchEEG)
**Difficulty:** Medium | **Impact:** Medium
```python
# Augment training data to simulate cross-subject variation
transforms = [GaussianNoise(), ChannelDropout(), FrequencyShift(), TemporalCrop()]
augmented_epochs = apply_augmentations(epochs, transforms)
```
- [ ] `shared/preprocessing/augmentation.py` — wrapper around TorchEEG transforms
- [ ] Configurable augmentation pipeline via YAML
- [ ] Applied only during training (not inference)
- [ ] Channel dropout simulates bad electrodes
- [ ] Frequency shift simulates IAF variation across subjects

### 5. Configuration
```yaml
# experiments/base_config.yaml
generalization:
  normalize_features: true       # ✅ already implemented
  use_specparam: true            # ✅ already implemented
  use_iaf: false                 # NEW — personalize band definitions
  use_alignment: false           # NEW — Euclidean Alignment
  use_ssl_pretrain: false        # NEW — selfEEG backbone
  use_augmentation: false        # NEW — TorchEEG transforms
  augmentation_transforms:       # NEW — which augmentations
    - gaussian_noise: {std: 0.1}
    - channel_dropout: {p: 0.1}
    - frequency_shift: {max_shift: 1.0}
```

## Acceptance Criteria
- [ ] All 4 techniques implemented as independent, pluggable modules
- [ ] Each can be enabled/disabled via YAML config
- [ ] All work with both encoding and addiction classifiers
- [ ] Documentation explains when to use each (small data → IAF+alignment, large data → SSL+augmentation)
- [ ] Benchmark showing accuracy improvement for each technique on UCI dataset

## Priority Order
1. IAF adjustment (easy, immediate win)
2. EEG augmentation (medium, improves CNN training)
3. Euclidean Alignment (medium, improves SVM/RF)
4. SSL pre-training (hard, biggest impact but needs large unlabeled data)

## Context
Identified during learning session on cross-subject generalization (concept 28). The building blocks (selfEEG, TorchEEG, MOABB) are already in our dependency list — just need wiring.
